### ISSUE TITLE
Crash when navigating back after opening any link on login flow

### ISSUE THREAD

**Issue by JorgeMucientes** [woocommerce/woocommerce-android#7578]
**Describe the bug**
Crash when navigating back after opening any link on login flow. 

**IMPORTANT NOTE** Only happens when the developer setting "Don't keep activities" is enabled

**To Reproduce**
Issue related with Android navigation changes added here #7534

**Screenshots**

https://user-images.githubusercontent.com/2663464/196935409-928035ad-a02c-47e7-9a69-d4249d16d5b1.mp4

**Expected behaviour**
Don't crash when navigating back from any of the opened links

**Stacktrace of crash**

```
FATAL EXCEPTION: main
                                                                                                    Process: com.woocommerce.android.dev, PID: 25206
                                                                                                    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.woocommerce.android.dev/com.woocommerce.android.ui.login.LoginActivity}: java.lang.ClassCastException: com.woocommerce.android.ui.login.overrides.WooLoginEmailFragment cannot be cast to androidx.navigation.fragment.NavHostFragment
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3449)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
                                                                                                    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loop(Looper.java:223)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7656)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
                                                                                                    Caused by: java.lang.ClassCastException: com.woocommerce.android.ui.login.overrides.WooLoginEmailFragment cannot be cast to androidx.navigation.fragment.NavHostFragment
                                                                                                    	at com.woocommerce.android.ui.login.LoginActivity.onCreate(LoginActivity.kt:180)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:7994)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:7978)
                                                                                                    	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3422)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601) 
                                                                                                    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                                    	at android.os.Looper.loop(Looper.java:223) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7656) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
```

**Comment by pachlava** [comment]
FWIW, for some reason does not happen to me on 10.8-rc-1, Samsung A31, Android 11.

**Comment by JorgeMucientes** [comment]
Thanks for sharing that @pachlava. Yes, that is expected because the changes that introduce this bug were merged in `v10.9` [here](https://github.com/woocommerce/woocommerce-android/pull/7534) 2 days ago.

 I'll fix this asap before we freeze this version.

**Comment by JorgeMucientes** [comment]
Additional note that this only happens when the developer setting "Don't keep activities" is enabled

**Comment by pachlava** [comment]
> that is expected because the changes that introduce this bug were merged in v10.9 [here](https://github.com/woocommerce/woocommerce-android/pull/7534) 2 days ago

Oh, sorry. I should have paid more attention to the issue description 🤦 . Thanks!
